astropy/io/fits/hdu/table.py

```diff
@@ -685,6 +693,10 @@ def _populate_table_keywords(self):
                 if val is not None:
                     keyword = keyword + str(idx + 1)
                     self._header[keyword] = val
+            for key in self._header_column.keys():
```

Comment:
Yes that is true. The Column class part seems a little difficult though. I'll need more time to figure out where to make changes.